### DISCUSSION TITLE
Unable to open RustDesk main window on client pc after updating to 1.2.7

### DISCUSSION THREAD

**Discussion by rec32** [rustdesk/rustdesk#8847] [General]
### Bug Description

I click on RustDesk tray icon -> "Open".
It launches 2 processes which close instantly:
RustDesk.exe "rustdesk:///"
RustDesk.exe --check-hwcodec-config
This happens on all 3 client pcs where i updated to 1.2.7.
On my own pc I'm able to open RustDesk.

### How to Reproduce

Connect to client pc.
Update RustDesk to 1.2.7.
Restart pc or don't, it doesn't matter.
Try to open RustDesk from tray or shortcut.

### Expected Behavior

RustDesk window opens.

### Operating system(s) on local side and remote side

Win 10 22H2 -> Win 10 22H2

### RustDesk Version(s) on local side and remote side

1.2.7 -> 1.2.7

### Screenshots

![Untitled](https://github.com/user-attachments/assets/0901baca-79dc-4b3c-b333-88a07f9c4a4e)

### Additional Context

I hope it's not a new "security feature"

**Comment by 21pages** [comment]
Does the --server process restart in a loop?

**Comment by rec32** [comment]
> Does the --server process restart in a loop?

No, nothing happens to it

Also forgot to mention i'm connecting directly and through lan (192.168.0.123:12345)

**Comment by 21pages** [comment]
Could you show the content of `%app%\Roaming\RustDesk\log\RustDesk_rCURRENT.log`? Remember to remove private information.

After that, please kill --server process, and it will auto restart, then try start main window. If it works, the reason is that --server does not start ipc correctly.

**Comment by rec32** [comment]
> [2024-07-25 17:12:11.497868 +03:00] INFO [src\core_main.rs:162] main start args:["rustdesk:///"]
> [2024-07-25 17:12:11.501967 +03:00] DEBUG [libs\hbb_common\src\config.rs:475] Configuration path: C:\Users\Admin\AppData\Roaming\RustDesk\config\RustDesk.toml
> [2024-07-25 17:12:11.502235 +03:00] DEBUG [libs\hbb_common\src\config.rs:475] Configuration path: C:\Users\Admin\AppData\Roaming\RustDesk\config\RustDesk2.toml

Killing --server process didn't help

**Comment by 21pages** [comment]
When I click tray, my log file is `[2024-07-25 22:19:21.516104 +08:00] INFO [src\core_main.rs:164] main start args:[]`, are you sure the program file is https://github.com/rustdesk/rustdesk/releases/download/1.2.7/rustdesk-1.2.7-x86_64.exe? Could you reinstall it?

**Comment by rec32** [comment]
Checked browser history. I downloaded exactly that file and installed(updated) it on my and client pcs.
Will try to reinstall.

**Comment by 21pages** [comment]
Remember to download from the link above, uninstall the old one completely and then install it.

**Comment by rec32** [comment]
Ok, to fix this I had to install AnyDesk, remove and reinstall RustDesk. Still, I don't think it was supposed to work this way.

**Comment by 21pages** [comment]
Does it work after reinstall? Something is wrong because the version before is not 1.2.7, the arg of 1.2.7 main window is not "rustdesk:///".

The previous reinstall was not successful and it is still running version 1.2.3 or 1.2.4. How did you reinstall it? I can't reproduce it after upgrade.

**Comment by rec32** [comment]
Yes it works fine after reinstall.
> The previous reinstall was not successful and it is still running version 1.2.3

I guess so. I was updating remotely, through RustDesk.
> How did you reinstall it?

Installed AnyDesk, deleted and reinstalled RustDesk, now it works.

**Comment by TempAccountNull** [comment]
For anyone else having this issue going forward, ensure you turn off `Only allow connection if RustDesk window is open` in settings window and hit OK.

**Reply by rec32** [reply]
I never had it enabled
